### ISSUE TITLE
Geocode evictions

### ISSUE THREAD

**Issue by SashaWeinstein** [NYCPlanning/db-equitable-development-tool#63]
The [residential evictions by federal marshals indicator](https://github.com/NYCPlanning/db-equitable-development-tool/issues/51) is the first indicator that requires geocoding. Hopefully the code written to geocode these records can be easily adopted to geocode other datasets

**Comment by SashaWeinstein** [comment]
The `63-Geocode-Evictions` branch should be merged into `51-Residential-Evictions-By-City-Marshals` before being merged into `dev`

**Comment by SashaWeinstein** [comment]
Not sure where the bash script to create the evictions indicator using geosupport/docker should live. Putting it in the `aggregate` folder for now. I think I remember Baiyue saying it's not great to organize files by language, don't have great idea of tradeoffs though

**Comment by SashaWeinstein** [comment]
### Set up .devcontainer
I've switched from a single `docker run` command to setting up a devcontainer. Want an environment in which I can run and debug the geocoding process.
I'm looking at how existing repos use geosupport via docker 

### Start with base python image and install geosupport
On facilities db, the Dockerfile sets up a python container and then installs geosupport on that container. I assumed that it would set up a container from the geosupport image instead. 

### Use geosupport image (not in for dev container)
Capital projects db on the other hand does use the geosupport docker image but just to run a single piece of code, not set up a development environment. PLUTO uses the image the same way. 

I'm not sure if I should try playing with the files in .devcontainer to set up a development environment using the docker geosupport image. Maybe there was a good reason facdb didn't do it this way. Will try for a bit and if it doesn't work then implement in similar way to facilities DB

**Comment by SashaWeinstein** [comment]
Think I'm getting a sense of the tradeoffs here. The image `mcr.microsoft.com/vscode/devcontainers/python` image used for facdb has python features like pylance and flake8 built in. If I want to use these I need to add instructions to dockerfile download them. There are may be other advantages to this image that I haven't come across yet. 

If I was on deadline I would take the implementation straight from facdb. But I think it's good to work through these issues one by one to get my docker proficiency up

**Comment by SashaWeinstein** [comment]
I have a geocoding process that assigns PUMAs to around 70% of the addresses that didn't have have location data in the source dataset. This brings the total percentage of records with a PUMA assigned to 97%. 
I examined the records that don't get geocoded with this process. Here are some things I observed

 - Misspellings lead to records not being geocoded. The first example I see is `1170 NAMKOKE STREET` with postal code 11691 which is clearly supposed to be 1170 *Nameoke* St. Geosupport returns an error that lists `['NAMEOKE AVENUE', 'NAMEOKE STREET']` as possible streets. So we could implement a solution where it tries to geocode one of the suggested similar streets.
 - Missing street types.  A good example is `31 WEST TREMONT`. Again geosupport returns suggestions, it gives `['WEST TREMONT AVENUE', 'WEST TREMONT-SEDGWICK AVENUE HSE']` as possible suggestions. We could implement a solution where it combines the street number of 31 with these street names and takes that hit
 - Issues with parsing address field. I use the `usaddress` package which is what facdb uses in it's geocoding. Unfortunately it's not that smart, it parsed `196 NOME AVENUE LEFT SIDE DOOR` into `[('196', 'AddressNumber'), ('NOME', 'StreetName'), ('AVENUE', 'StreetName'), ('LEFT', 'StreetName'), ('SIDE', 'StreetName'), ('DOOR', 'StreetName')]`. So it thinks left, side, and door are all plausible street names and doesn't identify "AVENUE" as the street type. So finding a better address parser could get our accuracy up
 - Extra punctuation: dashes, slashes, etc. `6908-8TH AVENUE` or `443/45 ST ANNS AVE`. We could implement something to deal with these
 
I'm not going to make any of these improvements now, my understanding is that 97% is good enough. I just want to document my observations if this issue is to be worked on more
